File: Africa 2010, Qualification Round/ProblemAStoreCredit.java
import java.io.FileNotFoundException;
import java.io.PrintWriter;
import java.util.Scanner;


public class ProblemAStoreCredit {

	public static void main(String[] args) throws FileNotFoundException{
		Scanner sc = new Scanner(System.in);
		PrintWriter file = new PrintWriter("output.txt");
		
		int numberOfCases = sc.nextInt();
		for(int i = 0; i < numberOfCases; i++){
			int credit = sc.nextInt();
			sc.nextInt(); //Number of items...Couldn't care less.
			sc.nextLine();
			String[] itemPrices = sc.nextLine().split(" ");
			String answer = "Case #"+(i+1)+": ";
			for(int j = 0; j < itemPrices.length; j++){
				for(int n = j; n < itemPrices.length; n++){
						if(Integer.parseInt(itemPrices[j]) + Integer.parseInt(itemPrices[n]) == credit && j != n){
							answer += (j+1)+" "+(n+1);
						}
					}
				}
			file.write(answer+"\n");
			}
		file.close();
		sc.close();
		}
	}

File: Africa 2010, Qualification Round/ProblemAStoreCredit.py
'''
Created on 01/08/2014

@author: miguel
'''

if __name__ == '__main__':

    def solve(itemPrices):
        for j in range(len(itemPrices)):
            for n in range(j, len(itemPrices)):
                if itemPrices[j] + itemPrices[n] == availableCredit and j != n:
                    return " " + str(j+1) + " " + str(n+1)

    
    numberOfCases = int(input())
    f = open("output.txt", 'w')
    for i in range(numberOfCases):
        
        availableCredit = int(input())
        input()
        itemPrices = map(int, input().split(" "))
        itemPrices = list(itemPrices)
        answer = "Case #"+str(i+1)+":"+solve(itemPrices)+'\n'
        f.write('%s\n' % answer)
    f.close()